### PR TITLE
Move host-disk to handler

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xpivarc** [comment]
/cc @davidvossel 
This is the thing we talked about. Let me know what you think.

File: pkg/virt-launcher/virtwrap/manager.go [hunk 1]
```diff
@@ -473,14 +472,6 @@ func (l *LibvirtDomainManager) preStartHook(vmi *v1.VirtualMachineInstance, doma
 		return domain, fmt.Errorf("preparing the pod network failed: %v", err)
 	}
 
-	// create disks images on the cluster lever
-	// or initialize disks images for empty PVC
-	hostDiskCreator := hostdisk.NewHostDiskCreator(l.notifier, l.lessPVCSpaceToleration, l.minimumPVCReserveBytes)
-	err = hostDiskCreator.Create(vmi)
-	if err != nil {
-		return domain, fmt.Errorf("preparing host-disks failed: %v", err)
-	}
```
Comment (davidvossel) [inline pkg/virt-launcher/virtwrap/manager.go:482]: if lessPVCSpaceToleration and minimumPVCReserveBytes are only used here, then there's a whole chain of things starting with template.go that can be removed related to these vars.

**Review by davidvossel — COMMENTED** [review]
why is moving all of this host-disk logic to handler better than having virt-handler simply set permissions on disks before performing the sync with virt-launcher?

**Comment by xpivarc** [comment]
> why is moving all of this host-disk logic to handler better than having virt-handler simply set permissions on disks before performing the sync with virt-launcher?

We would need to set permissions for disk(if exists) and for empty pvc (directory) because disks are created inside sync and not before. For me, it feels like additional code. Moving host-disk to the handler is simpler and it also allows us to drop the dependency from the launcher. This is good as the launcher has fewer privileges now.

**Comment by davidvossel** [comment]
> We would need to set permissions for disk(if exists) and for empty pvc (directory) because disks are created inside sync and not before

ah okay, so we potentially have to deal with the case where a disk doesn't exist as well. I think this makes sense to move to handler then

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

**Comment by enp0s3** [comment]
/hold
lets wait for https://github.com/kubevirt/kubevirt/pull/6280 to get in. We have a an issue with a sig-network macvtap test

**Pull request merged**